src/ucp/proto/proto_am.inl

```diff
@@ -70,10 +72,23 @@ ucs_status_t ucp_do_am_bcopy_multi(uct_pending_req_t *self, uint8_t am_id_first,
         packed_len = uct_ep_am_bcopy(uct_ep, am_id_last, pack_last, req, 0);
         UCS_PROFILE_REQUEST_EVENT_CHECK_STATUS(req, "am_bcopy_last", packed_len,
                                                packed_len);
-        return (packed_len < 0) ? packed_len : UCS_OK;
+        last = 1; //return (packed_len < 0) ? packed_len : UCS_OK;
     }
 
-    return (packed_len < 0) ? packed_len : UCS_INPROGRESS;
+    if (packed_len < 0) {
+        if (req->send.lane != req->send.pend_add_lane) {
+            // got NO_RES on a different lane than pending one
+            // resched
+            int ret = ucp_request_pending_add(req, &status);
+            ucs_assert(ret == 1);
```

Comment:
need to retry am_bcopy if we get "false" here

Reply:
done